NSrtm.Core/Pgm/PgmDataDescription.cs

```diff
@@ -0,0 +1,186 @@
+
+        public bool Equals(PgmDataDescription other)
+        {
+            return _offset.Equals(other._offset) && _scale.Equals(other._scale) && _originLat == other._originLat && _originLon == other._originLon &&
+                   _gridGraphWidthPoints == other._gridGraphWidthPoints && _gridGraphHeightPoints == other._gridGraphHeightPoints &&
+                   _maxValue == other._maxValue && _preambleLength == other._preambleLength;
+        }
+
+        #region Static Members
+
+        public static bool operator ==(PgmDataDescription left, PgmDataDescription right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(PgmDataDescription left, PgmDataDescription right)
+        {
+            return !left.Equals(right);
+        }
+
+        #endregion
+    }
+
+    internal static class PgmRawDataConverter
+    {
+        public static double ToEgmFormat(this UInt16 rawData, PgmDataDescription pgmParameters)
+        {
+            if (rawData > pgmParameters.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException("rawData");
+            }
+            return rawData * pgmParameters.Scale + pgmParameters.Offset;
+        }
+    }
+
+    internal static class PgmDataDescriptionExtractor
+    {
+        private static readonly Regex LatLonParser = new Regex("^(?<deg>[-+0-9]+)+(?<pos>[ENSW]*)$");
+
+        public static PgmDataDescription FromStream(Stream stream)
+        {
+            var preamble = extractPreambleFromStream(stream);
+            return getConstatantsFromPreamble(preamble);
+        }
+
+        private static string extractPreambleFromStream(Stream stream)
+        {
+            using (var reader = new StreamReader(stream, Encoding.UTF8, false, 1024, true))
+            {
+                const string _pgmMagicNumber = "P5";
+                var preamble = reader.ReadLine();
+                if (preamble != _pgmMagicNumber)
+                {
+                    throw new FileFormatException(String.Format("Wrong portable gray map magic number. Actual {0}, but should be {1}.", preamble, _pgmMagicNumber));
+                }
+
+                string lastLine;
+                do
+                {
+                    lastLine = reader.ReadLine();
+                    preamble += lastLine;
+                    preamble += "\n";
+                } while (lastLine.StartsWith("#", StringComparison.CurrentCulture));
+                preamble += reader.ReadLine();
+
+                return preamble;
+            }
+        }
+
+        private static PgmDataDescription getConstatantsFromPreamble(string preamble) //TODO define magic numbers
+        {
+            var unifyPreamble = preamble.Replace("#", "\n");
+            var words = unifyPreamble.Split(' ', '\n')
+                                     .ToList();
+            var offset = Convert.ToDouble(words[getParameterIndexUsingName(words, "Offset")], CultureInfo.InvariantCulture);
```

Comment:
Is it impossible to use methods from LINQ (like `First`) with lambda as filter instead of indexing using index found in separate method?